src/UserPreferences/Expenso.UserPreferences.Core/Application/Preferences/Read/Queries/GetPreference/GetPreferenceQueryHandler.cs

```diff
@@ -40,7 +40,7 @@ private PreferenceFilter GetFilter(GetPreferenceQuery query)
         (_, Guid? preferenceId, Guid? userId, bool? forCurrentUser, bool? includeFinancePreferences,
             bool? includeNotificationPreferences, bool? includeGeneralPreferences) = query;
 
-        if (forCurrentUser == true)
+        if (forCurrentUser is true)
```

Comment:
In that case:

src/BudgetSharing/Expenso.BudgetSharing.Infrastructure/Persistence/EfCore/Extensions/BudgetPermissionFilterExtensions.cs - It must be a '==' because it used  inside the expression three
`Expression<Func<BudgetPermission, bool>> predicate = p => p.Blocker == null || p.Blocker.IsBlocked == false;`

src/Api/Tests/Expenso.Api.Tests.E2E/Configuration/ExpensoWebApplication.cs - I will correct

Reply:
I can just correct it to:

```
public static class BudgetPermissionFilterExtensions
{
    public static Expression<Func<BudgetPermission, bool>> ToFilterExpression(this BudgetPermissionFilter filter)
    {
        Expression<Func<BudgetPermission, bool>> predicate = p => !IsBlocked(p);

        if (filter.Id is not null)
        {
            predicate = AndExpression<BudgetPermission>.And(leftExpression: predicate,
                rightExpression: x => x.Id == filter.Id);
        }

        if (filter.BudgetId is not null)
        {
            predicate = AndExpression<BudgetPermission>.And(leftExpression: predicate,
                rightExpression: x => x.BudgetId == filter.BudgetId);
        }

        if (filter.OwnerId is not null)
        {
            predicate = AndExpression<BudgetPermission>.And(leftExpression: predicate,
                rightExpression: x => x.OwnerId == filter.OwnerId);
        }

        if (filter.ParticipantId is not null)
        {
            predicate = AndExpression<BudgetPermission>.And(leftExpression: predicate,
                rightExpression: x =>
                    x.Permissions.Select(y => y.ParticipantId).Contains(PersonId.New(filter.ParticipantId.Value)));
        }

        if (filter.PermissionType is not null)
        {
            predicate = AndExpression<BudgetPermission>.And(leftExpression: predicate,
                rightExpression: x => x.Permissions.Select(y => y.PermissionType).Contains(filter.PermissionType));
        }

        return predicate;
    }

    private static bool IsBlocked(BudgetPermission budgetPermission) => budgetPermission.Blocker?.IsBlocked is true;
}
```

If it makes sense